x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/perform_bulk_action_route.ts

```diff
@@ -145,9 +203,43 @@ export const performBulkActionRoute = (
               },
               body: responseBody,
             });
+          case BulkAction.update:
+            processed = await chunkifyRulesAction(rules.data, async (rule) => {
```

Comment:
Should we also chunkify other bulk actions?

Reply:
yes, but I've decided to implement it in one of the subsequent PRs, where I will add on UI handling errors object in response, when any of rule updates fail. 
That way it won't break anything in UX. 

Another reasoning:
In current implementation of other bulk actions, any failure would end request, so chunking the requests could postpone showing the error to user.

Reply:
> I will add on UI handling errors object in response, when any of rule updates fail. That way it won't break anything in UX.

Not sure I understand. How sending requests in chunks could break UX?

> In current implementation of other bulk actions, any failure would end request, so chunking the requests could postpone showing the error to user

That should be fine. We can show an error when all requests have finished.

Reply:
> Not sure I understand. How sending requests in chunks could break UX?

Not sending in chunks, but returning 200 and errors array in response instead of failing the whole request. In my implementation chunking and error handling are coupled together in chunkify method.

> That should be fine. We can show an error when all requests have finished.

do you suggest to chunk requests for all actions, catch errors, and then if action is not `update` return error response?

Reply:
We should try making responses from different actions homogeneous. If one action returns 500 when any rule update fails, and the other returns 200 + an array of errors, we'll need additional logic on the front end to handle both cases. And to me, it looks like something we should avoid.

But do we need to respond with 200 if some of the rule updates fail?

Reply:
> We should try making responses from different actions homogeneous. If one action returns 500 when any rule update fails, and the other returns 200 + an array of errors, we'll need additional logic on the front end to handle both cases. And to me, it looks like something we should avoid.

Yes, I agree, response should be similar. But we don't have yet FE for update, so no additional logic is required for now 
 and won't be required in future as work on bulk editing continues. We can't have similar response for the rest of actions just now, as this would break current UX, which relies on error response when at least one of rules fail. 

Intention is to build `update` action API  in a way it can handle this case(we also have it in new UX, displaying partial failures), then when building UI for it, adjust API for the rest of actions. So:

1) we will not break current UX
2) will adjust API responses to one format for every action

> But do we need to respond with 200 if some of the rule updates fail?

I think we do, if from hundreds of updated rules only one fails, sending error status would be misleading. In my opinion for bulk actions we should return 200, list of errors + count of processed rules. This way we would be able to notify user how many failures happened and how many rules were processed.

Reply:
Discussed this topic via Zoom, so posting a summary here for visibility:
Responding with 200 OK when we have failed rules will require adding logic on FE. We already have error handling logic, to simplify it looks like the following:

```ts
try {
  await performAcrion()
} catch (e) {
  showError(e)
}
```

That logic doesn't cover cases when we receive errors with 200 OK. So we have to update it something like this:

```ts
try {
  const response = await performAcrion()
  if (response.errors) {
    showError(errors)
  }
} catch (e) {
  showError(e)
}
```

In my opinion, an extra `response.error` check is redundant. Moreover, from an API user perspective, it's pretty easy to overlook that 200 OK could contain errors and not handle that case at all. So we can avoid those pitfalls by returning 500 if any rule updates fail.

Reply:
changed it according to our discussion